File: backend/src/main/java/com/hrplatform/service/SkillService.java
package com.hrplatform.service;

import java.util.List;

import com.hrplatform.model.Skill;

public interface SkillService {

    Skill addSkill(Skill skill);

    Skill updateSkill(Long id, Skill skill);

    void deleteSkill(Long id);

    List<Skill> getAllSkills();

    Skill getSkillById(Long id);

    List<Skill> searchByName(String name);

    List<Skill> searchByCategory(String categoryName);

}

File: frontend/src/services/skillService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/skills';

export const getAllSkills = () => axios.get(API_URL);
export const getSkillById = (id) => axios.get(`${API_URL}/${id}`);
export const addSkill = (skill) => axios.post(API_URL, skill);
export const updateSkill = (id, skill) => axios.put(`${API_URL}/${id}`, skill);
export const deleteSkill = (id) => axios.delete(`${API_URL}/${id}`);
export const searchByName = (name) => axios.get(`${API_URL}/search?name=${name}`);
export const searchByCategory = (categoryName) => axios.get(`${API_URL}/category?categoryName=${categoryName}`);
export const addSkillToCandidate = (candidateId, skillId) =>
  axios.post(`http://localhost:8080/api/candidates/${candidateId}/skills/${skillId}`);

export const removeSkillFromCandidate = (candidateId, skillId) =>
  axios.delete(`http://localhost:8080/api/candidates/${candidateId}/skills/${skillId}`);